### PR TITLE
[TIMOB-25373] : iOS - Search result view of List did not close when w idow containing list view get closed

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-25373

### CODE REVIEW

File: iphone/Classes/TiUITableView.m [hunk 1]
```diff
@@ -312,6 +312,11 @@ - (id)init
 
 - (void)dealloc
 {
+  if ([searchController isActive]) {
+    searchController.view.hidden = YES;
+    [searchController setActive:NO];
```
Comment (hansemannn) [inline iphone/Classes/TiUITableView.m]: Why not remove it's delegates and nil it as well (like the others)?

Comment (vijaysingh-axway) [reply iphone/Classes/TiUITableView.m]: That is already there. See few lines below in this function :) . You will find -

    searchController.searchResultsUpdater = nil;
   searchController.delegate = nil;
   RELEASE_TO_NIL(searchController);

Comment (hansemannn) [reply iphone/Classes/TiUITableView.m]: I see this for the list-view, but not the table-view. Am I missing a snippet?

Comment (vijaysingh-axway) [reply iphone/Classes/TiUITableView.m]: I think so. You can see at line 338.

Comment (hansemannn) [reply iphone/Classes/TiUITableView.m]: I mean list-view, I cannot see nil'ing `searchResultsUpdater ` in `TiUIListView.m`.

Comment (vijaysingh-axway) [reply iphone/Classes/TiUITableView.m]: Ok. You are right. Done.

**Review by hansemannn — APPROVED** [review]

**Comment by hansemannn** [comment]
@ewieberappc Build is passing now! Merging here for consistency.

**Pull request merged**